### PR TITLE
chore: JWT warnings on failed JWT ops

### PR DESCRIPTION
**Pull request by peppelinux**
I'd like to have a warning meggase on failing JWT validations, or may use a debug for that and not a warning?

The messages must be improved.

actually with this PR we have

````
env/lib/python3.8/site-packages/idpyoidc/server/oidc/userinfo.py:191: KeyError
------------------------------------------------------------------------------------------------ Captured log call -------------------------------------------------------------------------------------------------
WARNING  cryptojwt.jwt:jwt.py:381 Exception unpacking a received signed or signed and encrypted Json Web Token for the issuer https://localhost:10000
WARNING  cryptojwt.jws.jws:jws.py:234 Exception "Invalid signature" caught
````

### CODE REVIEW

File: src/cryptojwt/jwt.py [hunk 1]
```diff
@@ -377,6 +377,10 @@ def unpack(self, token):
                 _msg_cls = self.iss2msg_cls[_info["iss"]]
             except KeyError:
                 _msg_cls = None
+                LOGGER.warning(
```
Comment (jschlyter) [inline src/cryptojwt/jwt.py]: If the KeyError is due to the fact that `_info['iss']` the exception handler will result in a new KeyError.

Comment (jschlyter) [reply src/cryptojwt/jwt.py]: Perhaps use the term "JWT" instead of "Json Web Token" :-)

Comment (peppelinux) [reply src/cryptojwt/jwt.py]: @jschlyter 

I'd prefer in this way

````
        # If I know what message class the info should be mapped into
        if self.msg_cls:
            _msg_cls = self.msg_cls
        else:
            _msg_cls = self.iss2msg_cls.get(_info["iss"], None)
            if not _msg_cls:
                LOGGER.warning(
                    "Exception unpacking a received JWT "
                    f"for the issuer {_info['iss']}"
                )
````

If you agree I can do this refactor

Comment (jschlyter) [reply src/cryptojwt/jwt.py]: yes, better!

**Comment by peppelinux** [comment]
did it, thank you

**Review by jschlyter — DISMISSED** [review]

File: src/cryptojwt/jws/jws.py [hunk 2]
```diff
@@ -226,14 +226,13 @@ def verify_compact_verbose(self, jws=None, keys=None, allow_none=False, sigalg=N
                 _key = key.key
 
             try:
-                if not verifier.verify(jwt.sign_input(), jwt.signature(), _key):
```
Comment (peppelinux) [inline src/cryptojwt/jws/jws.py:229]: @jschlyter 

I added also these

File: src/cryptojwt/jws/jws.py [hunk 3]
```diff
@@ -226,14 +226,13 @@ def verify_compact_verbose(self, jws=None, keys=None, allow_none=False, sigalg=N
...
-                    continue
-            except (BadSignature, IndexError):
-                pass
+                verifier.verify(jwt.sign_input(), jwt.signature(), _key)
```
Comment (jschlyter) [inline src/cryptojwt/jws/jws.py]: Why was the `continue` removed?

Comment (peppelinux) [reply src/cryptojwt/jws/jws.py]: I can't get a False but only an Exception if verify fails

then if True or False it will continue anyway, isn't so?

Comment (jschlyter) [reply src/cryptojwt/jws/jws.py]: [HMACSigner.verify](https://github.com/IdentityPython/JWTConnect-Python-CryptoJWT/blob/ca859033accf07bc707e8142cd5d623aabc57d9a/src/cryptojwt/jws/hmac.py#L33) returns `False` on failure

Comment (peppelinux) [reply src/cryptojwt/jws/jws.py]: I've just fixed it, sorry for the late!

**Comment by peppelinux** [comment]
@jschlyter finally got tests working well but py36 seems stucked :)

**Comment by jschlyter** [comment]
> @jschlyter finally got tests working well but py36 seems stucked :)

I think it is time to say goodbye to 3.6, see #128 - it was EOL over 6 months ago.

**Comment by peppelinux** [comment]
ok, so this is ready to be merged

**Review by rohe — APPROVED** [review]

**Review by jschlyter — COMMENTED** [review]
I'm not sure it is a good idea to log the entire JWT on signature failures. I'd prefer if we the caller decides to do that. If an attacker submits a huge JWT for processing, ending up with it in log files (as a warning) is suboptimal.

One could include the JWT in the exception if that helps the caller. In that case we can defined a `BadJwtSignature`  based on `BadSignature`  that has a jwt attribute.

**Comment by rohe** [comment]
Good comment!

**Comment by peppelinux** [comment]
Ok, this PR will take more time and design

don't block the next release for this, we can leave as it is with the pending work to be completed and have the next release with the other PRs that are good right now

**Pull request closed without merging**